drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,390 @@
+{
+	uint32_t status = REG_READ(RTC_CNTL_INT_ST_REG);
+
+	if (arg != NULL) {
+		const struct device *dev = arg;
+		struct esp32_touch_sensor_data *dev_data = dev->data;
+
+		if (dev_data->rtc_intr_msk & status) {
+			esp32_touch_sensor_interrupt_cb(arg);
+		}
+	}
+
+	REG_WRITE(RTC_CNTL_INT_CLR_REG, status);
+}
+
+static esp_err_t esp32_rtc_isr_install(intr_handler_t intr_handler, const void *handler_arg)
+{
+	esp_err_t err;
+
+	REG_WRITE(RTC_CNTL_INT_ENA_REG, 0);
+	REG_WRITE(RTC_CNTL_INT_CLR_REG, UINT32_MAX);
+	err = esp_intr_alloc(ETS_RTC_CORE_INTR_SOURCE, 0, intr_handler, (void *)handler_arg, NULL);
+
+	return err;
+}
+
+/**
+ * Handle debounced touch sensor touch state.
+ */
+static void esp32_touch_sensor_change_deferred(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct esp32_touch_sensor_channel_data *channel_data =
+			CONTAINER_OF(dwork, struct esp32_touch_sensor_channel_data, work);
+	const struct device *dev = channel_data->dev;
+	const struct esp32_touch_sensor_config *dev_cfg = dev->config;
+	int key_index = channel_data - &dev_cfg->channel_data[0];
+	const struct esp32_touch_sensor_channel_config
+		*channel_cfg = &dev_cfg->channel_cfg[key_index];
+
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	if (channel_data->last_status != channel_data->status) {
+#endif /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+		input_report_key(dev, channel_cfg->zephyr_code,
+				channel_data->status, true, K_FOREVER);
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+		channel_data->last_status = channel_data->status;
+	}
+#endif /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+}
+
+static int esp32_touch_sensor_init(const struct device *dev)
+{
+	struct esp32_touch_sensor_data *dev_data = dev->data;
+	const struct esp32_touch_sensor_config *dev_cfg = dev->config;
+	const int num_channels = dev_cfg->num_channels;
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	touch_hal_volt_t volt;
+
+	switch (dev_cfg->href_microvolt) {
+	case 2400000:
+		volt.refh = TOUCH_HVOLT_2V4;
+		break;
+	case 2500000:
+		volt.refh = TOUCH_HVOLT_2V5;
+		break;
+	case 2600000:
+		volt.refh = TOUCH_HVOLT_2V6;
+		break;
+	case 2700000:
+		volt.refh = TOUCH_HVOLT_2V7;
+		break;
+	}
+
+	switch (dev_cfg->lref_microvolt) {
+	case 500000:
+		volt.refl = TOUCH_LVOLT_0V5;
+		break;
+	case 600000:
+		volt.refl = TOUCH_LVOLT_0V6;
+		break;
+	case 700000:
+		volt.refl = TOUCH_LVOLT_0V7;
+		break;
+	case 800000:
+		volt.refl = TOUCH_LVOLT_0V8;
+		break;
+	}
+
+	switch (dev_cfg->lref_microvolt) {
+	case 0:
+		volt.atten = TOUCH_HVOLT_ATTEN_0V;
+		break;
+	case 500000:
+		volt.atten = TOUCH_HVOLT_ATTEN_0V5;
+		break;
+	case 1000000:
+		volt.atten = TOUCH_HVOLT_ATTEN_1V;
+		break;
+	case 1500000:
+		volt.atten = TOUCH_HVOLT_ATTEN_1V5;
+		break;
+	}
+
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
```

Comment:
maybe move this block in its own function?

Reply:
wouldn't DT_INST_ENUM_IDX() avoid all this code?

Reply:
@fabiobaltieri I'll follow @gmarull suggestion and this code will not be necessary.

Reply:
brilliant, even better